rest_framework/schemas/openapi.py

```diff
@@ -479,29 +493,67 @@ def _get_serializer(self, method, path):
                           .format(view.__class__.__name__, method, path))
             return None
 
-    def _get_request_body(self, path, method):
-        if method not in ('PUT', 'PATCH', 'POST'):
+    class SchemaMode(Enum):
+        RESPONSE = 1
+        BODY = 2
+
+    def _get_item_schema(self, serializer, schema_mode, method):
```

Comment:
I don't see the benefit of this refactoring. It's less clear than the separate methods, and doesn't save a lot of duplication (what with the enum and the bit `if...else` on the schema_mode...)

Reply:
You're right! I removed this refactoring.